### ISSUE TITLE
Dpnp run under dpctl with context throws error

### ISSUE THREAD

**Issue by reazulhoque** [IntelPython/dpnp#296]
The following reproducer produces the following output with Dpnp shipped with OneAPI beta 10:

Reproducer:
```
import dpnp
import dpctl

with dpctl.device_context("opencl:gpu"):
    result = dpnp.random.randn(10)
```

Output:
```
Running on: Intel(R) Graphics Gen9 [0x9bca]
DPCtrl SYCL queue used
SYCL kernels link time: 6.13809 (sec.)

terminate called after throwing an instance of 'cl::sycl::runtime_error'
  what():  Native API failed. Native API returns: -30 (CL_INVALID_VALUE) -30 (CL_INVALID_VALUE)
Aborted (core dumped)
```

The same code without dpctl seems to work:

Working Code:
```
import dpnp

result = dpnp.random.randn(10)`
```

Is this the expected behavior?

**Comment by shssf** [comment]
@reazulhoque Need to investigate. It is looks like it is not expected behavior. Thank you for the report.

**Comment by shssf** [comment]
@reazulhoque It looks like this is implemented. Could you please confirm we can close the issue?

**Comment by reazulhoque** [comment]
This issue is resolved in dpnp 4.0.
